prog/dftb+/lib_type/typegeometryhsd.F90

```diff
@@ -429,6 +417,231 @@ subroutine readTGeometryXyz_help(node, geo, text)
   end subroutine readTGeometryXyz_help
 
 
+  !> Reads the geometry from a node in a HSD tree in GEN format
+  subroutine readTGeometryVasp(node, geo)
+
+    !> Node containing the geometry in Gen format
+    type(fnode), pointer :: node
+
+    !> Contains the geometry on exit
+    type(TGeometry), intent(out) :: geo
+
+    type(string) :: text
+
+    call getFirstTextChild(node, text)
+    call readTGeometryVasp_help(node, geo, char(text))
+
+  end subroutine readTGeometryVasp
+
+
+  !> Helping routine for reading geometry from a HSD tree in VASP format
+  subroutine readTGeometryVasp_help(node, geo, text)
+
+    !> Node to parse (only needed to produce proper error messages)
+    type(fnode), pointer :: node
+
+    !> Contains the geometry on exit
+    type(TGeometry), intent(out) :: geo
+
+    !> Text content of the node
+    character(len=*), intent(in) :: text
+
+    type(string) :: txt
+    character(mc), allocatable :: vaspNames(:)
+    integer :: iStart, iOldStart, iErr, iEnd
+    integer :: ii, jj, iSp, iTmp
+    real(dp) :: coords(3), latVec(3), rTmp, rScale
+    integer, allocatable :: vaspSp(:)
+    integer, allocatable :: countSp(:)
+    type(listString) :: speciesNames
+    logical :: hasComment
+
+
+    ! Read `first' line of the POSCAR/CONTCAR file
+    ! This is actually a comment line, but contains by user convention the atomic symbols
+    ! In case it does not contain the atomic symbols, is empty or an HSD comment,
+    ! we have to ignore it, therefore we attempt to read it as `second' line first
+    hasComment = .false.
+    iStart = 1
+    iEnd = nextLine(text, iStart)
+    print'(a)', text(iStart:iEnd-1)
```

Comment:
Any particular reason for using print instead of write? I guess this is left over debugging anyway.

Reply:
Because print is easier to grep if one is searching for leftover debug printout. :)